api/src/services/generate-xform.js

```diff
@@ -0,0 +1,162 @@
+/**
+ * XForm generation service
+ * @module generate-xform
+ */
+const childProcess = require('child_process');
+const path = require('path');
+const logger = require('../logger');
+const db = require('../db');
+const formsService = require('./forms');
+
+const FORM_ROOT_OPEN = '<root xmlns:xf="http://www.w3.org/2002/xforms" xmlns:orx="http://openrosa.org/xforms" xmlns:enk="http://enketo.org/xforms" xmlns:kb="http://kobotoolbox.org/xforms" xmlns:esri="http://esri.com/xforms" xmlns:oc="http://openclinica.org/xforms" xmlns:h="http://www.w3.org/1999/xhtml" xmlns:ev="http://www.w3.org/2001/xml-events" xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:jr="http://openrosa.org/javarosa">';
+const MODEL_ROOT_OPEN = '<root xmlns="http://www.w3.org/2002/xforms" xmlns:xf="http://www.w3.org/2002/xforms" xmlns:h="http://www.w3.org/1999/xhtml" xmlns:ev="http://www.w3.org/2001/xml-events" xmlns:xsd="http://www.w3.org/2001/XMLSchema">';
+const ROOT_CLOSE = '</root>';
+const JAVAROSA_SRC = / src="jr:\/\//gi;
+
+const FORM_STYLESHEET = path.join(__dirname, '../xsl/openrosa2html5form.xsl');
+const MODEL_STYLESHEET = path.join(__dirname, '../../node_modules/enketo-xslt/xsl/openrosa2xmlmodel.xsl');
+
+const transform = (formXml, stylesheet) => {
+  return new Promise((resolve, reject) => {
+    const xsltproc = childProcess.spawn('xsltproc', [ stylesheet, '-' ]);
+    let stdout = '';
+    let stderr = '';
+    xsltproc.stdout.on('data', data => stdout += data);
+    xsltproc.stderr.on('data', data => stderr += data);
+    xsltproc.stdin.setEncoding('utf-8');
+    xsltproc.stdin.write(formXml);
+    xsltproc.stdin.end();
+    xsltproc.on('close', code => {
```

Comment:
I think you're right, it's a little more correct. I think because I'm not sharing streams it works out the same. Changed.

xsltproc cannot be found is actually handled by `close` and `exit`, but you're right I need to listen for `error` for more OS level failures. Added.